### ISSUE TITLE
Place to talk

### ISSUE THREAD

**Issue by ericsink** [chriswjones/sqlite-net-cipher-poc#1]
The first thing is to figure out if the right SQLitePCL.raw.dll is getting used.  Look for this file in your bin directory.  Compare it with the various ones available in the packages/SQLitePCL.raw directory.

I built your tree for iOS, and on that platform, it looks okay.  Not sure about Droid.

--
E

**Comment by chriswjones** [comment]
The SQLite.raw.dll in SnowReport/iOS/bin/iPhoneSimulator looks to be the same as the file in SnowReport/packages/SQLitePCL.raw.0.8.2/build/Xamarin.iOS10/pinvoke_packaged_sqlcipher/anycpu they are both 19.8 MB

**Comment by chriswjones** [comment]
I have been able to run on the simulator, it just looks like the sql cipher pragmas are not doing anything. you probably saw it already but in SnowReport/SnowReport.cs i create some connections and try to use the pragmas.

**Comment by ericsink** [comment]
Next is to try and figure out if the sqlite you are actually calling is in fact the one built in to SQLitePCL.raw.dll.

In your app, try running "SELECT sqlite_version();" and somehow get the result.  If you get 3.8.6, then you're probably talking to sqlcipher.  If you get something else, then you're probably talking to the system sqlite built-in to android.

**Comment by ericsink** [comment]
When you say "I have been able to run on the simulator", that kinda sounds like something there was successful?  But the rest of your sentence makes it sound like it was not.  Can you clarify why you said "I have been able to run on the simulator"?

**Comment by chriswjones** [comment]
I confirmed version 3.8.6.  The issue i am having is this:  when you run PRAGMA key = 'passphrase' it is my understanding that it encrypts the sqlite db with the key and that all subsequent connections must run that pragma as the first command to unlock it. 

Here is the issue:  I am opening a connection and running PRAGMA key = 'passme', but then the i try to open another connection and enter in the wrong password, but am still able to use the db on that connection when it should fail. maybe i am misinterpreting how to use sql cipher? this is where i am getting it from: https://www.zetetic.net/sqlcipher/sqlcipher-api/#key

**Comment by ericsink** [comment]
Nah, it sounds like you are doing it right, as best I can recall.  I haven't actually *used* sqlcipher myself in quite a while.

My questions were simply to confirm you actually have sqlcipher.  The behavior you are seeing is exactly what you would get if you were trying to use 'PRAGMA key' with plain sqlite.

It's possible that something in my bundling of sqlcipher has been screwed up.  I'll try to dig a little deeper.  Hopefully tomorrow.  :-)

**Comment by chriswjones** [comment]
Thanks for taking a look, i really appreciate it!!! and if there is anything i can do to help let me know

**Comment by ericsink** [comment]
I *think* the problem is that you have never actually written anything to the database file.

When I build your project and run it, I get the same results you do (no error on the second attempt to open the db file, with the wrong password).

And if I look at the mydb file in the simulator directory, it is 0 bytes.

OTOH, if I add a line to the first stanza:

conn.Execute ("CREATE TABLE foo (x int)");

right after the conn.Execute ("PRAGMA key = 'passme'");

now I get a mydb file which has something in it, and I get an error on the second attempt to open the file.

**Comment by chriswjones** [comment]
Hi Eric,

Sorry i have not gotten back to you sooner (the projects I am working on
have been absolutely crazy!).  You were correct and once I actually write
something to the DB everything is good to go!  I am extremely appreciative
of your help, and if there is ever anything I can do to assist you don't
hesitate to ask.

Thanks again,
Chris
